src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -259,19 +259,24 @@ impl Coordinator {
 
                     self.maybe_create_linked_cluster(source_id).await;
 
+                    let register_ts = self.get_local_write_ts().await.timestamp;
                     self.controller
                         .storage
-                        .create_collections(vec![(
-                            source_id,
-                            CollectionDescription {
-                                desc: source.desc.clone(),
-                                data_source,
-                                since: None,
-                                status_collection_id,
-                            },
-                        )])
+                        .create_collections(
+                            register_ts,
+                            vec![(
+                                source_id,
+                                CollectionDescription {
+                                    desc: source.desc.clone(),
+                                    data_source,
+                                    since: None,
+                                    status_collection_id,
+                                },
+                            )],
+                        )
                         .await
                         .unwrap_or_terminate("cannot fail to create collections");
+                    self.apply_local_write(register_ts).await;
```

Comment:
For `create_collections` calls that don't end up using the register_ts, we certainly don't need this `apply_local_write` call. It would be straightforward to make `create_collections` return a bool that indicates if it used the register_ts and then we could gate this behind that. Otoh, feels okay to apply a ts each time a user explicitly creates a source, mv, etc so not sure if the complexity is particularly worth it. Thoughts?

Reply:
Also, this path is for brand new collections, so they should always need the register_ts right? My understanding was that the register_ts is only not needed if the collection has already been registered.

Reply:
Hashed this out offline. I tried adding a bit more clarification to the `create_collections` rustdoc in this latest version, let me know if that would have helped the confusion.